bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/RuntimeUtils.java

```diff
@@ -197,6 +201,15 @@ public static String getMajorVersion(String version) {
         return version.split("\\.")[0];
     }
 
+    public static void validateObjectAssignabilityToBType(BMap<BString, Object> bMap) {
+        for (Object value : bMap.values()) {
+            if (value != null && !(value instanceof Number) && !(value instanceof BString) &&
```

Comment:
We can have a method in type checker with isValidBallerinaValue and use in createRecordvalue and here